modules/internal/ChapelBase.chpl

```diff
@@ -934,6 +934,10 @@ module ChapelBase {
 
   proc init_elts(x, s, type t, lo=0:s.type) : void {
 
+    if !isDefaultInitializable(t) then
```

Comment:
Hi @mppf, as part of this effort I felt that I needed to add a compiler warning to `init_elts` to prevent confusing warnings about default initializing bad tuples in internal module code. 

However this is breaking some methods like `list.toArray()` which default-initialize an array and then copy over the finished work over an `on` statement.

Do you have any ideas here?

Reply:
I've moved the check into `DefaultRectangular` instead of `init_elts`. One option for `list.toArray` is just to remove the `on` block.

Reply:
Yeah, I wouldn't put the check in init_elts unless you add a param about whether to error or not in these cases. In practice I would imagine it would be faster to adjust the compiler (see the work Vass did on errors for default-initialized arrays with non-default-initializable elements).

DefaultRectangular might be OK put I would expect we'd want `pragma "unsafe"` to disable the error; which is why doing it in the compiler might be necessary.